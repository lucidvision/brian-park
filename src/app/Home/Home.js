import React, { Component, Fragment } from 'react'
import styled from 'styled-components'
import { WOW } from 'wowjs'
import {
  SectionIntro,
  SectionSkills,
  SectionTechnologies,
  SectionWork,
  SectionContact
} from './index'
import { media } from 'styles'

const StyledMain = styled.main`
  overflow-x: hidden;

  & > * {
    padding: 14rem 10rem;

    ${media.desktop`
      padding: 18rem 10rem;
    `}

    ${media.tabLand`
      padding: 12rem 8rem;
    `}

    ${media.tabPort`
      padding: 12rem 4rem;
    `}

    ${media.phone`
      padding: 8rem 2rem;
    `}
  }
`

class Home extends Component {
  componentDidMount () {
    new WOW({
      live: false
    }).init()
  }
  render () {
    return (
      <Fragment>
        <header>
          <SectionIntro />
        </header>
        <StyledMain>
          <SectionSkills />
          <SectionTechnologies />
          <SectionWork />
        </StyledMain>
        <footer>
          <SectionContact />
        </footer>
      </Fragment>
    )
  }
}

export default Home
