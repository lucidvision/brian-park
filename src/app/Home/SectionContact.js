import React from 'react'
import styled from 'styled-components'
import { BarSocialMedia } from 'components'

const StyledSection = styled.section`
  display: flex;
  justify-content: center;
`

export default function SectionContact (props) {
  return (
    <StyledSection>
      <BarSocialMedia />
    </StyledSection>
  )
}
