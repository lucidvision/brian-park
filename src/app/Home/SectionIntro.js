import React from 'react'
import styled from 'styled-components'
import { HeadingPrimary, ImageCircle, Paragraph } from 'components'
import brian from 'assets/brian.jpg'

const StyledSection = styled.section`
  text-align: center;
`

export default function SectionIntro (props) {
  return (
    <StyledSection>
      <ImageCircle imagePath={brian} />
      <HeadingPrimary text={'Hi, I\'m Brian'} />
      <Paragraph primary text='I have a passion for bringing products to market - from product management to software engineering.' />
    </StyledSection>
  )
}
