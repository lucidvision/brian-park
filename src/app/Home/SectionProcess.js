import React from 'react'

export default function SectionProcess (props) {
  return (
    <div>SectionProcess</div>
  )
}
