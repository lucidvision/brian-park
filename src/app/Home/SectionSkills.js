import React from 'react'
import styled from 'styled-components'
import { HeadingSecondary, Skill } from 'components'
import { media, utilities } from 'styles'
import _ from 'lodash'

const StyledSection = styled.section`
  margin-top: -10rem;

  ${media.phone`
    margin-top: -4rem;
  `}
`

const StyledDiv = styled.div`
  display: flex;
  justify-content: space-between;

  ${media.phone`
    flex-direction: column;
    align-items: center;

    > * {
      &:not(:last-child) {
        margin-bottom: ${utilities.marginMedium};
      }
    }
  `}
`

const StyledHeadingSecondary = styled(HeadingSecondary)`
  text-align: center;
  margin-bottom: ${utilities.marginLarge};

  ${media.phone`
    margin-bottom: ${utilities.marginMedium};
  `}
`

const StyledSkill = styled(Skill)`
  width: 30%;

  ${media.phone`
    width: 90%;
  `}
`

const skills = [
  {
    title: 'Strategy',
    content: 'A big goal requires efficient planning. I use the 80/20 rule and iterative development to deliver exceptional results.'
  },
  {
    title: 'Product',
    content: 'Customer obsession inspires elegant design. I use feedback to craft interfaces that are easy to use.'
  },
  {
    title: 'Technology',
    content: 'Optimization drives continuous improvement. I use cutting-edge frameworks and tools to write clean, scalable code.'
  }
]

export default function SectionSkills (props) {
  return (
    <StyledSection>
      <StyledHeadingSecondary text='What I do' />
      <StyledDiv>
        {_.map(skills, (skill, index) => <StyledSkill key={index} {...skill} />)}
      </StyledDiv>
    </StyledSection>
  )
}
