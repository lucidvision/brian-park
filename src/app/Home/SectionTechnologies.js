import React from 'react'
import styled from 'styled-components'
import Technologies from './Technologies'
import { HeadingSecondary, Paragraph } from 'components'
import { colors, media, utilities } from 'styles'

const StyledSection = styled.section`
  background-image: linear-gradient(
    to right bottom,
    ${colors.gradientLight},
    ${colors.gradientDark});
  background-size: cover;
  transform: skewY(-7deg);

  & > * {
    transform: skewY(7deg);
  }
`

const StyledHeadingSecondary = styled(HeadingSecondary)`
  text-align: center;
  margin-bottom: ${utilities.marginLarge}

  ${media.phone`
    margin-bottom: ${utilities.marginMedium}
  `}
`

const StyledDiv = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  ${media.phone`
    flex-direction: column;
  `}
`

const StyledParagraph = styled(Paragraph)`
  max-width: 50rem;

  ${media.phone`
    margin-bottom: ${utilities.marginMedium}
    text-align: center;
  `}
`

export default function SectionTechnologies (props) {
  return (
    <StyledSection>
      <StyledHeadingSecondary color={colors.white} text='Technologies' />
      <StyledDiv>
        <StyledParagraph color={colors.white} text={'My primary focus is JavaScript. JavaScript\'s ubiquity makes it arguably the most productive programming language. It can be used for web, mobile and APIs. It has cutting-edge tools, libraries and frameworks. Its first-class functions allow for a declarative programming style. JavaScript empowers me to bring products to market fast.'} />
        <Technologies />
      </StyledDiv>
    </StyledSection>
  )
}
