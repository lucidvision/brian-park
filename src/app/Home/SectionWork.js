import React from 'react'
import styled from 'styled-components'
import { HeadingSecondary, ImageCoastline, Paragraph } from 'components'
import { media, utilities } from 'styles'

const StyledHeadingSecondary = styled(HeadingSecondary)`
  text-align: center;
  margin-bottom: 1rem;

  ${media.phone`
    margin-bottom: ${utilities.marginMedium};
  `}
`

const StyledDiv = styled.div`
  display: flex;
  justify-content: space-between;
  margin-right: -4rem;

  ${media.phone`
    flex-direction: column;
    align-items: center;
    margin-right: 0;
  `}
`

const StyledDivInfo = styled.div`
  margin-top: 8rem;

  ${media.tabLand`
    margin-top: 6rem;
  `}

  ${media.tabPort`
    margin-top: 4rem;
  `}

  ${media.phone`
    margin-top: 0;
    text-align: center;
  `}
`

const StyledImageCoastline = styled(ImageCoastline)`
  height: 50%;
  width: 50%

  ${media.phone`
    height: 100%;
    width: 100%;
  `}
`

export default function SectionWork (props) {
  return (
    <section>
      <StyledHeadingSecondary text='Coastline Market' />
      <StyledDiv>
        <StyledDivInfo>
          <Paragraph text={'As CTO, I led the development of Coastline\'s "Harvest-on-Demand" platform, providing e-commerce and logistics automation for the seafood industry. The technology includes a mobile app, a web app and a web-services API.'} /><br />
          <Paragraph text={'Coastline uses cross-platform JavaScript powered by React, React-Native and Express to respond rapidly to customer feedback. Coastline also leverages powerful technologies like Stripe, SendGrid, Twilio, Routific, Heroku and Firebase to maximize its value proposition.'} /><br />
          <Paragraph text={'Through an iterative development process, Coastline delights customers with accurate design and product-market fit.'} /><br />
        </StyledDivInfo>
        <StyledImageCoastline />
      </StyledDiv>
    </section>
  )
}
