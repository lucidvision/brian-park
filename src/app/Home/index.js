import Home from './Home'
import SectionIntro from './SectionIntro'
import SectionContact from './SectionContact'

export default Home
export {
  SectionIntro,
  SectionContact
}
