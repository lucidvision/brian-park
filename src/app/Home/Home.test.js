import React from 'react'
import { shallow } from 'enzyme'
import Home from './Home'

describe('Home', () => {
  const home = shallow(<Home />)

  it('renders correctly', () => {
    expect(home).toMatchSnapshot()
  })

  it('renders the intro section', () => {
    expect(home.find('SectionIntro').exists()).toBeTruthy()
  })

  it('renders the skills section', () => {
    expect(home.find('SectionSkills').exists()).toBeTruthy()
  })

  it('renders the technologies section', () => {
    expect(home.find('SectionTechnologies').exists()).toBeTruthy()
  })

  it('renders the work section', () => {
    expect(home.find('SectionWork').exists()).toBeTruthy()
  })

  it('renders the contact section', () => {
    expect(home.find('SectionContact').exists()).toBeTruthy()
  })
})
