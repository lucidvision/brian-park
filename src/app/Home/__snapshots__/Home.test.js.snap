// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<React.Fragment>
  <header>
    <SectionIntro />
  </header>
  <styled.main>
    <SectionSkills />
    <SectionTechnologies />
    <SectionWork />
  </styled.main>
  <footer>
    <SectionContact />
  </footer>
</React.Fragment>
`;
